File: android/app/src/main/java/com/eva/app/vehicle/VehicleProfile.kt
// android/app/src/main/java/com/eva/app/vehicle/VehicleProfile.kt
package com.eva.app.vehicle

import kotlinx.serialization.Serializable

/**
 * Kullanıcının aracına dair, cihazda şifreli olarak saklanan profil.
 * Sıfır-PII: bu veri bir kişiyi değil bir ARACI tanımlar, sunucuya
 * yalnızca soket tipi/batarya kapasitesi gibi filtre parametreleri olarak
 * gönderilir (bkz. VoiceQueryRequest.vehicleConnectorTypes) — plaka, VIN
 * numarası gibi gerçek dünya kimlikleri BU MODELDE YOK ve asla eklenmemeli.
 *
 * currentChargePercent: DÜRÜSTLÜK NOTU — bu alan gerçek bir araç
 * telemetrisi (OBD-II / üretici API'si) entegrasyonundan GELMİYOR, çünkü
 * böyle bir entegrasyon henüz projede yok. Kullanıcı bunu manuel girer
 * (bkz. ChargeLevelUpdateDialog, ileride eklenecek). Dashboard'da "68%"
 * gösteriliyorsa bu kullanıcının en son girdiği değerdir, canlı bir sensör
 * okuması değildir — bunu UI metninde de netleştireceğiz.
 */
@Serializable
data class VehicleProfile(
    val brand: String,
    val model: String,
    val batteryCapacityKwh: Double,
    val connectorType: String, // "CCS2", "TESLA_NACS", "TYPE2" vb. — backend enum'uyla birebir
    val currentChargePercent: Int = 80,
    val estimatedRangeMiles: Int? = null,
) {
    init {
        require(batteryCapacityKwh > 0) { "Batarya kapasitesi pozitif olmalı" }
        require(currentChargePercent in 0..100) { "Şarj yüzdesi 0-100 aralığında olmalı" }
    }

    val displayName: String
        get() = "$brand $model"
}

/**
 * Onboarding ekranında kullanıcıya gösterilecek yaygın soket tipi
 * seçenekleri — backend'deki connector_type enum'uyla (database/schema.sql)
 * BİREBİR aynı string değerleri kullanır, aksi halde /v1/stations/nearby
 * filtresi sessizce hiçbir sonuç döndürmez.
 */
enum class ConnectorOption(val backendValue: String, val displayLabel: String) {
    CCS2("CCS2", "CCS2 (Avrupa Standardı)"),
    CCS1("CCS1", "CCS1 (Kuzey Amerika)"),
    TESLA_NACS("TESLA_NACS", "Tesla (NACS)"),
    CHADEMO("CHAdeMO", "CHAdeMO"),
    TYPE2("TYPE2", "Type 2 (AC)"),
}

File: ios/EvaApp/EdgeAI/Battery/VehicleProfile.swift
// ios/EvaApp/EdgeAI/Battery/VehicleProfile.swift
import Foundation

/// Araca özgü, tamamen cihaz üstünde saklanan profil.
/// Sıfır-PII: bu veri hiçbir zaman sunucuya ham olarak gönderilmez.
struct VehicleProfile: Codable, Equatable {
    let modelIdentifier: String
    let batteryCapacityKwh: Double
    let usableBatteryRatio: Double
    let baseConsumptionKwhPer100Km: Double
    let regenEfficiencyRatio: Double
    let maxDcChargeRateKw: Double
    let supportedConnectorTypes: [String]
    let preconditioningPenaltyKwh: Double

    init(
        modelIdentifier: String,
        batteryCapacityKwh: Double,
        usableBatteryRatio: Double,
        baseConsumptionKwhPer100Km: Double,
        regenEfficiencyRatio: Double,
        maxDcChargeRateKw: Double,
        supportedConnectorTypes: [String],
        preconditioningPenaltyKwh: Double = 0.8
    ) {
        precondition(batteryCapacityKwh > 0, "Batarya kapasitesi pozitif olmalı")
        precondition(usableBatteryRatio > 0 && usableBatteryRatio <= 1, "usableBatteryRatio (0,1] aralığında olmalı")

        self.modelIdentifier = modelIdentifier
        self.batteryCapacityKwh = batteryCapacityKwh
        self.usableBatteryRatio = usableBatteryRatio
        self.baseConsumptionKwhPer100Km = baseConsumptionKwhPer100Km
        self.regenEfficiencyRatio = regenEfficiencyRatio
        self.maxDcChargeRateKw = maxDcChargeRateKw
        self.supportedConnectorTypes = supportedConnectorTypes
        self.preconditioningPenaltyKwh = preconditioningPenaltyKwh
    }

    var usableCapacityKwh: Double {
        batteryCapacityKwh * usableBatteryRatio
    }
}
